syntax = "proto3";

service RsyncNotifications {
  rpc TakeSnapshotAfterRsyncCompletion (RsyncCompletionRequest) returns (SnapshotCompletionResponse);
  rpc PrepareDatasetBeforeRsyncStart (RsyncStartRequest) returns (PreparationCompletionResponse);
  rpc ListSnapshots (ListSnapshotsRequest) returns (ListSnapshotsResponse);
}

message RsyncCompletionRequest {
  string folder_name = 1;
}

message SnapshotCompletionResponse {
  bool success = 1;
  string message = 2;
}

message RsyncStartRequest {
  string folder_name = 1;
}

message PreparationCompletionResponse {
  bool success = 1;
  string message = 2;
}

message ListSnapshotsRequest {
  string folder_name = 1; // Optional argument
}

message ListSnapshotsResponse {
  bool success = 1;
  string message = 2;
  repeated string snapshots = 3;
}